Expand ci build & verify across all packages

tom--pollard opened the pull request:
Closes #77 

There may be a preference for having the lint commit before instead of after the commit in which the feature is enabled in CI, as such each commit passes. We may also want to tweak the `golangci.yml` file before committing the lint changes, as I don't know the provenance of it and it may require tailoring for the entire codebase.

Qinusty commented on .github/workflows/ci.yml:
```diff
@@ -4,7 +4,7 @@ on:
   pull_request:
 jobs:
-  verify-terraform:
+  terraform-check:
```
NIT: We should avoid bundling these changes with the lint commit

tom--pollard replied:
Agreed, missed this on the rebase

Qinusty commented on feeds/feed.go:
```diff
@@ -9,11 +10,13 @@ import (
 
 const schemaVer = "1.0"
 
+var errCutoff = errors.New("package was created before cutoff time")
```
Can this be const?

tom--pollard replied:
I believe the convention is to use package level vars for wrapping errors with `Errorf` https://github.com/Djarvur/go-err113#reports

Qinusty commented on feeds/nuget/nuget_test.go:
```diff
@@ -13,15 +13,19 @@ import (
+	srv := testutils.HTTPServerMock(handlers)
+	testEndpoint, err = url.Parse(srv.URL)
+	if err != nil {
+		fmt.Println("Unexpected error during test url parsing: %w", err)
```
In a test this should probably be `t.Fatalf()`

Qinusty commented on feeds/nuget/nuget_test.go:
```diff
@@ -58,38 +62,61 @@ func indexMock(w http.ResponseWriter, r *http.Request) {
-	_, _ = w.Write([]byte(response))
+	_, err := w.Write([]byte(response))
+	if err != nil {
+		fmt.Println("Unexpected error during mock http server write: %w", err)
```
In a test this should probably be `t.Fatalf()` (duplicate)

Qinusty commented on feeds/nuget/nuget_test.go:
```diff
@@ -58,38 +62,61 @@ func indexMock(w http.ResponseWriter, r *http.Request) {
-	_, _ = w.Write([]byte(response))
+	_, err := w.Write([]byte(response))
+	if err != nil {
+		fmt.Println("Unexpected error during mock http server write: %w", err)
```
In a test this should probably be `t.Fatalf()` (duplicate)

Qinusty commented on feeds/nuget/nuget_test.go:
```diff
@@ -58,38 +62,61 @@ func indexMock(w http.ResponseWriter, r *http.Request) {
-	_, _ = w.Write([]byte(response))
+	_, err := w.Write([]byte(response))
+	if err != nil {
+		fmt.Println("Unexpected error during mock http server write: %w", err)
```
In a test this should probably be `t.Fatalf()` (duplicate)

Qinusty commented on feeds/nuget/nuget_test.go:
```diff
@@ -58,38 +62,61 @@ func indexMock(w http.ResponseWriter, r *http.Request) {
-	_, _ = w.Write([]byte(response))
+	_, err := w.Write([]byte(response))
+	if err != nil {
+		fmt.Println("Unexpected error during mock http server write: %w", err)
```
In a test this should probably be `t.Fatalf()` (duplicate)

Qinusty commented on feeds/nuget/nuget_test.go:
```diff
@@ -58,38 +62,61 @@ func indexMock(w http.ResponseWriter, r *http.Request) {
+	}
 }
 
 func makeTestURL(suffix string) string {
-	path, _ := url.Parse(suffix)
+	path, err := url.Parse(suffix)
+	if err != nil {
+		fmt.Println("Unexpected error during test url parsing: %w", err)
```
In a test this should probably be `t.Fatalf()` (duplicate)

Qinusty commented on feeds/packagist/packagist_test.go:
```diff
@@ -40,19 +43,60 @@ func TestFetch(t *testing.T) {
+		e.g. https:\/\/packagist.org\/metadata\/changes.json?since=16145146222715 for example."
+		,"timestamp":16145146222715}`))
+		if err != nil {
+			fmt.Println("Unexpected error during mock http server write: %w", err)
```
In a test this should raise the HTTP error similarly to the implementation of `HttpServe` outside of the tests

Qinusty commented on feeds/packagist/packagist_test.go:
```diff
@@ -40,19 +43,60 @@ func TestFetch(t *testing.T) {
-			_, _ = w.Write([]byte(response))
+			_, err := w.Write([]byte(response))
+			if err != nil {
+				fmt.Println("Unexpected error during mock http server write: %w", err)
```
In a test this should raise the HTTP error similarly to the implementation of HttpServe outside of the tests (duplicate)

Qinusty commented on feeds/packagist/packagist_test.go:
```diff
@@ -40,19 +43,60 @@ func TestFetch(t *testing.T) {
+	"time":1614513806},{"type":"update","package":"ossf/package","time":1614514502},
+	{"type":"update","package":"ossf/package~dev","time":1614514502}],"timestamp":16145145025048}`))
+	if err != nil {
+		fmt.Println("Unexpected error during mock http server write: %w", err)
```
In a test this should raise the HTTP error similarly to the implementation of HttpServe outside of the tests (duplicate)

Qinusty commented on feeds/pypi/pypi_test.go:
```diff
@@ -71,4 +72,7 @@ func updatesXmlHandle(w http.ResponseWriter, r *http.Request) {
 </rss>
 `))
+	if err != nil {
+		fmt.Println("Unexpected error during mock http server write: %w", err)
```
In a test this should raise the HTTP error similarly to the implementation of HttpServe outside of the tests (duplicate)

Qinusty commented on feeds/rubygems/rubygems_test.go:
```diff
@@ -119,4 +115,7 @@ func rubyGemsPackagesResponse(w http.ResponseWriter, r *http.Request) {
 
 `))
+	if err != nil {
+		fmt.Println("Unexpected error during mock http server write: %w", err)
```
In a test this should raise the HTTP error similarly to the implementation of HttpServe outside of the tests (duplicate)

tom--pollard commented:
@Qinusty I think I've addressed your comments in regards to raising `http.Error` / subsequent `t.Fatalf` in places where error handling was added, what do you think?

Qinusty reviewed (approved):
LGTM!
